W-16688568 Add MUnit phase 2 docs 1024 cc (#372)
mulesoft/docs-code-builder: modules/ROOT/pages/int-test-munit.adoc

[[open-test]]
== Open MUnit Tests

Open the test suite XML from the *Testing* panel or the *Explorer*.

* *Testing panel* - Open the *Testing* panel from the activity bar in the IDE:
//open project (with tests) step as local variable
. {open-project} 
. Click image:anypoint-code-builder::icon-munit-testing.png["",18,18] (*Testing*) to open the *Testing* panel.

> . Click or mouse over the suite or test name, and click image:anypoint-code-builder::icon-munit-go-to-test-gray.png["",18,18] or highlighted image:anypoint-code-builder::icon-munit-go-to-test.png["",18,18] (*Go to test*) to open the XML file that contains the test.

Reviewer: Should you have (*Go to test*) for the first image as well? Or alt text. Think about how a screenreader will read this.

Author: Fixed!

+
--
[calloutlist]

> .. All suites: To run all tests in all suites, click the *Run Tests*, *Debug Tests*, or *Run Tests with Coverage* icon.

Reviewer: I assume there's a code sample that this is referring to? :> ETA - I see the local variable now.

Author: Yeah there are code samples.

+
image:anypoint-code-builder::munit-xml-test-run-panel-xml.png["Running test in the Test panel and XML"]
+

> After the test runs, the panel and XML indicate the result.

Reviewer: ```suggestion
After the test runs, the panel and XML show the result.
```

//TODO: VERIFY STEP, ANYTHING TO ADD?
. Examine the data, and address any issues.

By default, running tests fail after a timeout period. If a debugging session reaches a breakpoint in a test, the test and the build fail at timeout. See xref:munit::munit-test-concept.adoc#test-timeout-attribute[Test TimeOut Attribute].

//
//RUN: COVERAGE
[[run-coverage]]
== Run MUnit Tests with Coverage

> Run your test with coverage to measure how much of a Mule application has been executed by a set of MUnit tests.

Reviewer: ```suggestion
Run your test with coverage to measure how much of a Mule application is executed by a set of MUnit tests.
```

> //open project (with tests) step as local variable

Reviewer: This is a weird use for local variables...but hey, I guess it works!

Author: Yeah this came from Duke. It's pretty cool.

. {open-project}
. Run your test with coverage using one of the following options:

* From the *Testing* panel, click *Run Test with Coverage* at any level (all tests, all tests in a suite, or a single test).
+
For more guidance, see <<run-test-panel>>.

> * From the suite XML, use the right-click menu to run the test.

Reviewer: Can you say what the option is? Or are there a bunch?

Author: There could be many.

> * From the suite XML, use the right-click menu to run the test.

Reviewer: ```suggestion
* From the suite XML, right-click and select an option to run the test.
```

//TODO: VERIFY STEP, ANYTHING TO ADD?
. Examine the data, and address any issues.

By default, running tests fail after a timeout period. If a debugging session reaches a breakpoint in a test, the test and the build fail at timeout. See xref:munit::munit-test-concept.adoc#test-timeout-attribute[Test TimeOut Attribute].

//
//RUN: COVERAGE
[[run-coverage]]
== Run MUnit Tests with Coverage

Run your test with coverage to measure how much of a Mule application has been executed by a set of MUnit tests.

> This procedure requires the Enterprise Edition of Mule runtime engine (Mule). For licensing information, see xref:mule-runtime::installing-an-enterprise-license.adoc[Install an Enterprise License].

Reviewer 2: we should remove this line, because this is transparent to users now. MUnit automatically starts an enterprise runtime using the runtime installed in acb

After <<run, running a test>>, fix any errors.

> . Review test errors in the XML by clicking a failing test's name from the *Testing* panel.

Reviewer: ```suggestion
. Review test errors in the XML by clicking the name of the failing test from the *Testing* panel.
```

* To rerun the test from the *Testing* panel, click image:anypoint-code-builder::icon-munit-run-test.png["",18,18] (*Run test*). 

> * To rerun the test from the Testing Results panel, open the panel, click the failing test in the test logs, and click image:anypoint-code-builder::icon-munit-run-test.png["",18,18] (*Run test*) from the test logs.

Reviewer: ```suggestion
* To rerun the test from the Testing Results panel, open the panel, click the name of the failing test in the test logs, and click image:anypoint-code-builder::icon-munit-run-test.png["",18,18] (*Run test*) from the test logs.
```

Create an MUnit test suite in your project from the *Explorer*. 

. Open your project to `{munit-src}` in the *Explorer*.
. Create a suite file in the `munit` folder:

> .. Right-click on the `munit` folder and select *New MUnit Test Suite File...*.

Reviewer: ```suggestion
.. Right-click the `munit` folder and select *New MUnit Test Suite File...*.
```
